src/cache/abstract-assignment-cache.ts

```diff
@@ -19,11 +26,18 @@ export function assignmentCacheKeyToString({ subjectKey, flagKey }: AssignmentCa
   return getMD5Hash([subjectKey, flagKey].join(';'));
 }
 
-export function assignmentCacheValueToString({
-  allocationKey,
-  variationKey,
-}: AssignmentCacheValue): string {
-  return getMD5Hash([allocationKey, variationKey].join(';'));
+export function assignmentCacheValueToString(cacheValue: AssignmentCacheValue): string {
+  const fieldsToHash: string[] = [];
+
+  if ('allocationKey' in cacheValue && 'variationKey' in cacheValue) {
+    fieldsToHash.push(cacheValue.allocationKey, cacheValue.variationKey);
+  }
+
+  if ('banditKey' in cacheValue && 'actionKey' in cacheValue) {
+    fieldsToHash.push(cacheValue.banditKey, cacheValue.actionKey);
+  }
+
+  return getMD5Hash(fieldsToHash.join(';'));
```

Comment:
nit: this code looks like it's trying to cram two functions into one which makes it convoluted to follow (e.g., is it possible that both `if`s match? none?)

If we add new event types (or decide to add `allocation`/`variation` key to bandit action), this is going to get messy.

A cleaner solution would be to make cache type generic over what it caches and/or accept key/value mapping functions as parameters.

Reply:
+1 to this comment

Reply:
I don't think we could do generic because we only want to hash a specific subset of the provided properties. The code is indeed just handling two specific possible code paths

Reply:
Update: since the other fields are part of the cache key and we know they will be unique, we can just simply hash the entire value, making this generic 💪